src/MultiLogging.hs

```diff
@@ -198,4 +198,4 @@ runLogFuncMulti :: LoggingConstraints m => ML.LogT L.AppName m () -> m ()
 runLogFuncMulti logFunc = do
   AppNameLoggers loggers <- askLoggers
   mapM_ runLoggerOver loggers
-  where runLoggerOver logger = ML.runLogTSafe logger logFunc
+  where runLoggerOver logger = ML.runLogT' logger logFunc
```

Comment:
I didn't notice hangup since I switched from Readline to Haskeline. Is this another problem ?

About what you say:

> logging should be fail-safe: failure to log must not result in application failures

Actually I wonder if I agree with that. I think I like that when something that should work stop working, that the application just crashes very loudly so a proper action can be taken. An application health can be monitored separately (e.g. by doing HTTP GETs on some route periodically). So we would see 5xx instead of silently stop logging anything and possibly never notice it.